applications/fishing-map/src/features/report/ReportLayerPanel.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import cx from 'classnames'
+import { useTranslation } from 'react-i18next'
+import { TagList, Tooltip } from '@globalfishingwatch/ui-components'
+import { TagItem } from '@globalfishingwatch/ui-components/dist/tag-list'
+import { DatasetTypes } from '@globalfishingwatch/api-types'
+import { UrlDataviewInstance } from 'types'
+import styles from 'features/workspace/shared/LayerPanel.module.css'
+import { getFlagsByIds } from 'utils/flags'
+import { getSchemaFieldsSelectedInDataview } from 'features/datasets/datasets.utils'
+import {
+  getSourcesSelectedInDataview,
+  isFishingDataview,
+  isPresenceDataview,
+} from 'features/workspace/heatmaps/heatmaps.utils'
+
+type LayerPanelProps = {
+  index: number
+  dataview: UrlDataviewInstance
+}
+
+function ReportLayerPanel({ dataview, index }: LayerPanelProps): React.ReactElement {
+  const { t } = useTranslation()
+  const sourcesOptions: TagItem[] = getSourcesSelectedInDataview(dataview)
+  const nonVmsSources = sourcesOptions.filter((source) => !source.label.includes('VMS'))
+  const vmsSources = sourcesOptions.filter((source) => source.label.includes('VMS'))
+  let mergedSourceOptions: TagItem[] = []
+  if (vmsSources?.length > 1) {
+    mergedSourceOptions = [
+      ...nonVmsSources,
+      {
+        id: 'vms-grouped',
+        label: `VMS (${vmsSources.length} ${t('common.country_plural', 'countries')})`,
+        tooltip: vmsSources.map((source) => source.label).join(', '),
+      },
+    ]
+  }
+
+  const fishingFiltersOptions = getFlagsByIds(dataview.config?.filters?.flag || [])
+  const gearTypesSelected = getSchemaFieldsSelectedInDataview(dataview, 'geartype')
+  const fleetsSelected = getSchemaFieldsSelectedInDataview(dataview, 'fleet')
+  const vesselsSelected = getSchemaFieldsSelectedInDataview(dataview, 'vessel_type')
```

Comment:
Same than above here, encapsulating:
```js
const gearTypesSelected = getSchemaFieldsSelectedInDataview(dataview, 'geartype')
```
 and the repeated piece of jsx above:
```jsx
gearTypesSelected.length > 0 && (
      <div className={styles.filter}>
        <label>{t('layer.gearType_plural', 'Gear types')}</label>
        <TagList
          tags={gearTypesSelected}
          color={dataview.config?.color}
          className={styles.tagList}
        />
      </div>
    )
```

In something like:
```jsx
<DatasetSchemaField dataview={dataview} field="geartype" />
```

Reply:
I agree 👍